docs/enhancements/cancel-day-2-host-installation.md

```diff
@@ -0,0 +1,181 @@
+- Host cancellation should happen within a reasonable amount of time in order to support dynamic scaling 
+
+### Non-Goals
+
+- Hosts that are installed in a day-1 cluster is out of scope of this enhancement
+- Cluster installation cancellation is out of scope
+- The way reclaim/unbind works should not be affected by these changes
+- This enhancement is focused solely on Assisted-Service offerings using Kubernetes API (SaaS should not be affected by this)
+- The host should still be responsive, in other words, any hosts that are unreachable are out scope of this proposal
+
+## Proposal
+
+Allow cancelling an installing day-2 host through Assisted-Service's Kubernetes API.
+
+### User Stories
+
+#### Story 1
+
+As a cluster admin, when I create a hosted cluster using Hypershift, I'd like to bind and install worker hosts into my cluster. 
+However, if I want to remove the host from the cluster before they finish installing, I should be able to do so and reuse the host for another cluster.
+
+#### Story 2
+
+As a cluster admin using Hypershift, I want to use the autoscaler.
+When the scaling happens and hosts are bound/unbound from the cluster, the hosts should bind and unbind 
+easily from the cluster.
+
+#### Story 3
+
+As a cluster admin, if I install a day-2 host to a cluster, and I see the host installation fails from
+inside the host. I should be able to cancel the installation once it starts and quickly reset the host.
+
+### Implementation Details/Notes/Constraints [optional]
+
+There is an already existing cancel host installation API which maps to the unbind and reclaim stages.
+This is currently only being used when a day-1 cluster installation is cancelled. 
+
+#### Detection of Cancellation
+
+When an Agent's cluster deployment reference is removed and the host is in the middle of installation,
+the controller will consider that a trigger to cancel the installation.
+
+Currently, there is a webhook that prevents the cluster reference from being removed
+when a host is in the middle of installation. The proposal for this enhancement is to remove
+that webhook to unblock the removal of the cluster reference.
+
+#### Cancelling Installation
+
+The host should transition to cancelled state.
+
+The `stop-installation` command is sent to a host that is in the middle of installing.
+The command will be sent with retries in case there is a scenario such as the host just starting rebooting.
+
+However, if the command doesn't receive a successful reply, the controller will assume
+the host is unreachable and transition the host to a pending-user-action state.
+
+#### Post Cancellation
+
+##### Successful Cancellation
+
+If a host cancellation was successful, the host will be transitioned to unbinding then reclaiming.
+
+During reclaim, the host should be rebooted with the discovery ISO and its state should be reset to
+discovering once the host tries to register again. This already exists with the current reclaim flow
```

Comment:
Just a reminder that this will only happen for hosts that have a BMH or have been successfully installed.

If a host never joins the cluster after reboot and does not have a BMH the user will need to manually boot it into the discovery ISO.

I think that's fine (as there's nothing else we can do), but I just wanted to make a note here because it sounds like we're always expecting the host to go through the actual reclaim process.

Reply:
Thank you for mentioning this!! I actually didn't know this. Will add some language about this